src/gf2_solver.cpp

```diff
@@ -40,12 +40,12 @@ namespace green::mbpt {
     //
     MPI_Datatype dt_matrix     = utils::create_matrix_datatype<std::complex<double>>(_nso * _nso);
     MPI_Op       matrix_sum_op = utils::create_matrix_operation<std::complex<double>>();
-    _coul_int_c_1              = new df_integral_t(_path, _nao, _NQ, _bz_utils);
-    _coul_int_c_2              = new df_integral_t(_path, _nao, _NQ, _bz_utils);
-    _coul_int_c_3              = new df_integral_t(_path, _nao, _NQ, _bz_utils);
-    _coul_int_c_4              = new df_integral_t(_path, _nao, _NQ, _bz_utils);
-    _coul_int_x_3              = new df_integral_t(_path, _nao, _NQ, _bz_utils);
-    _coul_int_x_4              = new df_integral_t(_path, _nao, _NQ, _bz_utils);
+    _coul_int_c_1              = new df_integral_t(_path, _nao, _NQ, _bz_utils, _verbose);
+    _coul_int_c_2              = new df_integral_t(_path, _nao, _NQ, _bz_utils, _verbose);
+    _coul_int_c_3              = new df_integral_t(_path, _nao, _NQ, _bz_utils, _verbose);
+    _coul_int_c_4              = new df_integral_t(_path, _nao, _NQ, _bz_utils, _verbose);
+    _coul_int_x_3              = new df_integral_t(_path, _nao, _NQ, _bz_utils, _verbose);
+    _coul_int_x_4              = new df_integral_t(_path, _nao, _NQ, _bz_utils, _verbose);
```

Comment:
this may possibly lead to trouble. We're currently using memory heuristics to load the integrals into shared mem usiing 1/2 of total memory. This will create 6 different shmem areas each allowed to use 1/2 of total mem.

Not your fault – I missed that. Propose to put a comment before merging this one in.

Reply:
Also I haven't checked how the 'corrections' are done. Need to rework and revisit, possibly once Sergei revisits the integrals.